utils/splunk/splunk_telemetry_base.py

```diff
@@ -71,7 +71,17 @@ def _dispatch_and_await_search(self, instance, saved_searches):
 
         for saved_search in saved_searches:
             try:
+                # a unique saved search key to persist the data
+                persist_status_key = instance.instance_config.base_url + saved_search.name
+                if self.status.data.get(persist_status_key) is not None:
+                    for (sid, saved_search) in self.status.data[persist_status_key]:
+                        res_code = instance.splunkHelper.finalize_sid(sid, saved_search)
+                        if res_code == 200:
+                            self.update_persistent_status(instance.instance_config.base_url, saved_search.name,
+                                                          (sid, saved_search), 'remove')
```

Comment:
So, when we fail to clean up a saved search, we still continue. This is not the idea. We only want to continue when all saved searches are cleaned up